crates/core/src/transport/connection_handler.rs

```diff
@@ -656,9 +674,15 @@ impl<S: Socket> UdpPacketsListener<S> {
                                 direction = "inbound",
                                 "Failed to establish gateway connection"
                             );
+                            // Only block peers with incompatible protocol versions, not other errors.
+                            // Issue #2292: Previously, ALL ConnectionEstablishmentFailure errors
+                            // caused the peer to be blocked for 10 minutes. This line was buggy:
+                            //   cause.starts_with("remote is using a different protocol version");
+                            // It evaluated to a boolean but was discarded - the insert always ran.
                             if let TransportError::ConnectionEstablishmentFailure { cause } = error {
-                                cause.starts_with("remote is using a different protocol version");
-                                outdated_peer.insert(remote_addr, Instant::now());
+                                if cause.starts_with("remote is using a different protocol version") {
```

Comment:
this stuff is very britle, I think we need to move to better typed error catching so we can distinguish properly